home/home1.1/main.rs

```diff
@@ -17,8 +17,8 @@ fn main() {
         let mut has_negative = false;
         let mut largest_negative: Option<f64> = None;
         let mut valid_input = true;
-        for i in list.split_whitespace() {
-            match i.trim().parse::<f64>() {
+        for num_str in input_list.split_whitespace() {
```

Comment:
It is better to use filter_map function something like the following one
let negative_max_value = input_list.split_whitespace()
       .filter_map(|s| s.parse::<f64>().ok())
       .filter_map(|val| val < 0)
        .max();

usage of such syntax is more readable as we can understand intention without reading for loops and if statements.